### PR TITLE
Add a general PR template

### PR DESCRIPTION
**Pull request by Renny4Real**
This adds a generic PR template to help us structure of PR descriptions in a way that would help reviewers understand the context of the changes being introduced.

### CODE REVIEW

**Review by humulla — APPROVED** [review]
I like this idea!

File: pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+### *What changes have we introduced*
+
+Add descriptions of what changes were made to address the problem and provide a solution.
+
+Include any links to commits that introduce significant additions of code if they help explain the process of coming to the solution. e.g Addition of test database setup, addition of first test and production class, removal of buggy code, etc.
+
+#### _Checklist_
+
+- [ ] Code builds correctly
+- [ ] Added tests to cover new production code added as necessary
```
Comment (elcorbs) [inline pull_request_template.md:22]:
```suggestion
- [ ] Added tests to cover all new production code
```

File: pull_request_template.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+- [ ] All tests passing
```
Comment (elcorbs) [inline pull_request_template.md:23]: This feels a given with the point above and also enforced by the pipeline.

File: pull_request_template.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+- [ ] Added comments the README / documentation, as necessary
+- [ ] Refactored code as neccessary
```
Comment (elcorbs) [inline pull_request_template.md:25]: Maybe something like "checked all code for possible refactoring"?

File: pull_request_template.md [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+Add a summary of the problem and any additional context that helps explain what the issue we're trying to solve is.
+
+### *What changes have we introduced*
+
+Add descriptions of what changes were made to address the problem and provide a solution.
+
+Include any links to commits that introduce significant additions of code if they help explain the process of coming to the solution. e.g Addition of test database setup, addition of first test and production class, removal of buggy code, etc.
+
+#### _Checklist_
```
Comment (elcorbs) [inline pull_request_template.md:19]: Can we have a point about having an integration test if necessary, please

**Review by elcorbs — APPROVED** [review]
Love this

File: pull_request_template.md (hunk 4 above)
Comment (Renny4Real) [reply pull_request_template.md:19]: Yep, will add that in 👍

File: pull_request_template.md (hunk 2 above)
Comment (Renny4Real) [reply pull_request_template.md:23]: Yeah, that makes sense. I can remove this line.

**Pull request merged**